lib/logflare/endpoints/cache.ex

```diff
@@ -102,6 +100,11 @@ defmodule Logflare.Endpoints.Cache do
 
     unless state.disable_cache, do: refresh(proactive_querying_ms(state))
 
+    # register on syn
+    pid = self()
+    :syn.register(:endpoints, {query.id, params}, pid)
```

Comment:
out of curiosity, why don't you register the process with {:via, :syn, SOME_ID}?

Reply:
wanted to put all the syn code together for readability, since it would have to join the group too.
placing it as a via tuple is fine as well, no issues with that.